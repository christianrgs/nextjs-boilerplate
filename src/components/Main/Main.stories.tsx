import Main from './Main'
import type { ComponentStory, ComponentMeta } from '@storybook/react'

export default {
  title: 'Components/Main',
  component: Main,
  args: {
    title: 'title template',
    description: 'description template'
  }
} as ComponentMeta<typeof Main>

export const Template: ComponentStory<typeof Main> = args => <Main {...args} />

export const Basic = Template.bind({})
Basic.args = {
  title: 'title basic',
  description: 'description basic'
}
