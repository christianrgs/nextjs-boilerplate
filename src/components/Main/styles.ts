import styled from 'styled-components'

export const Wrapper = styled.main`
  width: 100%;
  height: 100%;
  display: flex;
  padding: 3rem;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  text-align: center;
  background-color: #06092b;
  color: #ffffff;
`

export const Title = styled.h1`
  font-size: 2.5rem;
`

export const Description = styled.h2`
  font-size: 2rem;
  font-weight: 400;
`
