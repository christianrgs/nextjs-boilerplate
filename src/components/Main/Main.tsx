import * as Styled from './styles'

const Main = () => {
  return (
    <Styled.Wrapper>
      <Styled.Title>Nextjs Boilerplate</Styled.Title>
      <Styled.Description>
        TypeScript, React, Next.js, Styled Components, Jest and React Testing Library
      </Styled.Description>
    </Styled.Wrapper>
  )
}

export default Main
