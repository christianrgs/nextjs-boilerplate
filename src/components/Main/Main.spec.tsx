import { screen, render } from '@testing-library/react'
import Main from './Main'

describe('Main', () => {
  it('should render Main', () => {
    const { container } = render(<Main />)

    expect(
      screen.getByRole('heading', { level: 1, name: /nextjs boilerplate/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the container background color correctly', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toHaveStyle({ 'background-color': '#06092b' })
  })
})
