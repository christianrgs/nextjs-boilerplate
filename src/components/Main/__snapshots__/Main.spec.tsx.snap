// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render Main 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 3rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  background-color: #06092b;
  color: #ffffff;
}

.c1 {
  font-size: 2.5rem;
}

.c2 {
  font-size: 2rem;
  font-weight: 400;
}

<main
  class="c0"
>
  <h1
    class="c1"
  >
    Nextjs Boilerplate
  </h1>
  <h2
    class="c2"
  >
    TypeScript, React, Next.js, Styled Components, Jest and React Testing Library
  </h2>
</main>
`;
